src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/HologramProjector.java

```diff
@@ -119,14 +147,59 @@ private void openEditor(@Nonnull Player p, @Nonnull Block projector) {
+        menu.addItem(1, new CustomItemStack(Material.CLOCK, "&7Offset X: &e" + getXOffset(projector), "", "&fLeft Click: &7+0.1", "&fRight Click: &7-0.1"));
+        menu.addMenuClickHandler(1, (pl, slot, item, action) -> {
+            double offsetX = NumberUtils.reparseDouble(getXOffset(projector) + (action.isRightClicked() ? -0.1F : 0.1F));
+            double offsetY = getYOffset(projector);
+            double offsetZ = getZOffset(projector);
+
+            if (offsetX < -MAX_HORIZONTAL_OFFSET) {
+                offsetX = -MAX_HORIZONTAL_OFFSET;
+            } else if (offsetX > MAX_HORIZONTAL_OFFSET) {
+                offsetX = MAX_HORIZONTAL_OFFSET;
+            }
+            
+            ArmorStand hologram = getArmorStand(projector, true);
+            Location l = new Location(projector.getWorld(), projector.getX() + offsetX, projector.getY() + offsetY, projector.getZ() + offsetZ);
+            hologram.teleport(l);
+
+            BlockStorage.addBlockInfo(projector, OFFSET_PARAMETER_X, String.valueOf(offsetX));
+            openEditor(pl, projector);
+            return false;
+        });
+
+        menu.addItem(1, new CustomItemStack(Material.CLOCK, "&7Offset Y: &e" + getYOffset(projector) + 1.0D, "", "&fLeft Click: &7+0.1", "&fRight Click: &7-0.1"));
         menu.addMenuClickHandler(1, (pl, slot, item, action) -> {
-            double offset = NumberUtils.reparseDouble(Double.valueOf(BlockStorage.getLocationInfo(projector.getLocation(), OFFSET_PARAMETER)) + (action.isRightClicked() ? -0.1F : 0.1F));
+            double offsetX = getXOffset(projector);
+            double offsetY = NumberUtils.reparseDouble(getYOffset(projector) + (action.isRightClicked() ? -0.1F : 0.1F));
+            double offsetZ = getZOffset(projector);
+            
+            ArmorStand hologram = getArmorStand(projector, true);
+            Location l = new Location(projector.getWorld(), projector.getX() + offsetX, projector.getY() + offsetY, projector.getZ() + offsetZ);
+            hologram.teleport(l);
+
+            BlockStorage.addBlockInfo(projector, OFFSET_PARAMETER, String.valueOf(offsetY));
+            openEditor(pl, projector);
+            return false;
+        });
+
+        menu.addItem(1, new CustomItemStack(Material.CLOCK, "&7Offset Z: &e" + getZOffset(projector), "", "&fLeft Click: &7+0.1", "&fRight Click: &7-0.1"));
+        menu.addMenuClickHandler(1, (pl, slot, item, action) -> {
+            double offsetX = getXOffset(projector);
+            double offsetY = getYOffset(projector);
+            double offsetZ = NumberUtils.reparseDouble(getZOffset(projector) + (action.isRightClicked() ? -0.1F : 0.1F));
+
+            if (offsetZ < -MAX_HORIZONTAL_OFFSET) {
+                offsetZ = -MAX_HORIZONTAL_OFFSET;
+            } else if (offsetZ > MAX_HORIZONTAL_OFFSET) {
+                offsetZ = MAX_HORIZONTAL_OFFSET;
+            }
```

Comment:
Math.max and Math.min should be used here, there might also be a method to do both